bin/update-ips.bash

```diff
@@ -15,231 +15,351 @@ source "${here}/common.bash"
 
 CHECK_CLUSTER="${1}" # sc, wc or both
 DRY_RUN=true
-if [[ "${2}" == "update" ]]; then
-    DRY_RUN=false
+if [[ "${2}" == "apply" ]]; then
+  DRY_RUN=false
 fi
 has_diff=0
 
 # Compares a list IPs with the list of IPs in the yaml path and file specified and returns the diff return code.
 # If DRY_RUN is set, it will output to stdout, otherwise it will just return the diff return code silently.
 # Usage: diffIPs <yaml_path> <file> <IP 1> <IP ..>
 diffIPs() {
-    local yaml_path
-    local file
-    local IPS
-    yaml_path="${1}"
-    file="${2}"
-    shift 2
-    IPS=("$@")
-    tmp_file=$(mktemp --suffix=.yaml)
-
-    yq4 -n '. = []' > "${tmp_file}"
-    for ip in "${IPS[@]}"; do
-        yq4 -i '. |= . + ["'"${ip}"'/32"]' "${tmp_file}"
-    done
+  local yaml_path="${1}"
+  local file="${2}"
+  local IPS
+  shift 2
+  IPS=("$@")
+  tmp_file=$(mktemp --suffix=.yaml)
+
+  yq4 -n '. = []' >"${tmp_file}"
+  for ip in "${IPS[@]}"; do
+    yq4 -i '. |= . + ["'"${ip}"'/32"]' "${tmp_file}"
+  done
 
-    if $DRY_RUN; then
-        out_file=/dev/stdout
-    else
-        out_file=/dev/null
-    fi
-    diff -U3 --color=always \
-        --label "${file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${yaml_path}"' // [] | sort_by(.)' "${file}") \
-        --label expected <(yq4 -P '. | sort_by(.)' "${tmp_file}") > "${out_file}"
-    DIFF_RETURN=$?
-    rm "${tmp_file}"
-    return ${DIFF_RETURN}
+  if $DRY_RUN; then
+    out_file=/dev/stdout
+  else
+    out_file=/dev/null
+  fi
+  diff -U3 --color=always \
+    --label "${file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${yaml_path}"' // [] | sort_by(.)' "${file}") \
+    --label expected <(yq4 -P '. | sort_by(.)' "${tmp_file}") >"${out_file}"
+  DIFF_RETURN=$?
+  rm "${tmp_file}"
+
+  return ${DIFF_RETURN}
 }
```

Comment:
Simplification

```bash
diff_cidrs() {
  local config_key="${1}"
  local config_file="${2}"
  local cidrs=("${@:3}")

  if $DRY_RUN; then
    out_file=/dev/stdout
  else
    out_file=/dev/null
  fi

  # use diff return implicitly
  diff -U3 --color=always \
    --label "${config_file//${CK8S_CONFIG_PATH}\//}" <(yq4 -P "${config_key}"' // [] | sort' "${config_file}") \
    --label expected <(yq4 -P 'split(" ") | sort' <<< "${cidrs[@]}") > "${out_file}"
}
```

We should not modify anything in the input at this stage, since the input to the diff should reflect the input into update else the diff will be misleading.